aliyun/src/test/java/org/apache/iceberg/aliyun/oss/mock/AliyunOSSMockRule.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aliyun.oss.mock;
+
+import com.aliyun.oss.OSS;
+import com.aliyun.oss.OSSClientBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.apache.iceberg.aliyun.oss.AliyunOSSTestRule;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AliyunOSSMockRule implements AliyunOSSTestRule {
+  private static final Logger LOG = LoggerFactory.getLogger(AliyunOSSMockRule.class);
+
+  private final Map<String, Object> properties;
+
+  private AliyunOSSMockApp ossMockApp;
+
+  private AliyunOSSMockRule(Map<String, Object> properties) {
+    this.properties = properties;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  @Override
+  public void start() {
+    ossMockApp = AliyunOSSMockApp.start(properties);
+  }
+
+  @Override
+  public void stop() {
+    ossMockApp.stop();
+  }
+
+  @Override
+  public OSS createOSSClient() {
+    String endpoint = String.format("http://localhost:%s", properties.getOrDefault(
+        AliyunOSSMockApp.PROP_HTTP_PORT,
+        AliyunOSSMockApp.PORT_HTTP_PORT_DEFAULT));
+    return new OSSClientBuilder().build(endpoint, "foo", "bar");
+  }
+
+  @Override
+  public String keyPrefix() {
+    return "mock-objects/";
+  }
+
+  private File rootDir() {
+    Object rootDir = properties.get(AliyunOSSMockApp.PROP_ROOT_DIR);
+    Preconditions.checkNotNull(rootDir, "Root directory cannot be null");
+    return new File(rootDir.toString());
+  }
+
+  @Override
+  public void setUpBucket(String bucket) {
+    createOSSClient().createBucket(bucket);
+  }
+
+  @Override
+  public void tearDownBucket(String bucket) {
+    try {
+      Files.walk(rootDir().toPath())
+          .filter(p -> p.toFile().isFile())
```

Comment:
Why if we have a bucket named `test-bucket`, and have a object named `path/to/a.dat`,  Then the full local path will be `<root-dir>/test-bucket/path/to/a.dat`,   will we remove the `<root-dir>/test-bucket/path/to` directory also ?

Reply:
In aliyun OSS,  the `<root-dir>/test-bucket/path/to` won't be a specific directory in the real production, but in our local OSS storage app,  it will be a real directory (though it's not a real object that people could see by using aliyun OSS SDK).

Reply:
Regarding to the scenario you list above,  the directories are deleted in the deleteBucket. However,  we indeed have problem in another scenario. if we create an object with name "aa/bb/cc.txt", and then remove it. after that we create a new object with name "aa/bb". it has problem because of deleting cc.txt in current logic only. I will fix it in separate PR.